In [1]:
!pip install selenium

In [2]:
import selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.common.exceptions import NoSuchElementException,StaleElementReferenceException
import warnings
warnings.filterwarnings('ignore')
import time

In [100]:
#I'm using Quikr website for scraping used car data
driver=webdriver.Chrome(r'C:\Users\Admin\Downloads\chromedriver_win32 (1)\chromedriver.exe')
driver.get('https://www.quikr.com/cars/used+cars+all-india+z1399')

In [64]:
driver.maximize_window()

In [65]:
driver.execute_script('window.scrollBy(0,1500)')

In [101]:
#Scraping the name of the car's
name=[]
for i in driver.find_elements(By.XPATH,'//div[@class="car-info__footer"]'):
    name.append(i.text.split('\n')[0].split('-')[0])
name

['Maruti Suzuki Eeco ',
 'Audi A4 2.0 TDI 143bhp ',
 'Hyundai Santro Xing ',
 'Mahindra Marazzo M2 8 STR ',
 'Hyundai Venue ',
 'Skoda Kodiaq ',
 'Hyundai Fluidic Verna 1.6 VTVT SX ',
 'Mahindra XUV 500 W6 ',
 'Mahindra TUV300 T6 ',
 'BMW 3 Series 320d Highline ',
 'Maruti Suzuki Swift ZXI ABS ',
 'Toyota Etios G ',
 'Maruti Suzuki Alto 800 Lxi ',
 'Hyundai Grand i10 magna 1.2 crdi ',
 'Toyota Innova 2.5 G BS III 8 STR ',
 'Maruti Suzuki Swift Dzire Tour VDi ',
 'Tata Nexon XZA Plus Diesel ',
 'Mitsubishi Pajero Sport 2.5 MT ',
 'Datsun RediGo ',
 'Maruti Suzuki Swift Dzire ',
 'Hyundai Grand i10 ',
 'Hyundai i10 ',
 'Maruti Suzuki Swift VDi ABS BS IV ',
 'Renault Kwid RXT ',
 'Nissan Terrano XV 110 DIESEL ',
 'Mahindra Jeep Classic ',
 'Honda City VX ',
 'Maruti Suzuki Eeco 5 STR ',
 'Honda City 1.5 S MT ',
 'Honda Jazz 1.2 S MT ',
 'Maruti Suzuki Baleno ZETA 1.2 K12 CVT ',
 'Maruti Suzuki Swift Dzire ZDI AMT ',
 'Hyundai i10 Era ',
 'Maruti Suzuki Swift Dzire LDi ',
 'Land Rover Rang

In [102]:
#Scraping the manufacturing_year of the car's
manufacturing_year=[]
for i in driver.find_elements(By.XPATH,'//div[@class="car-info__footer"]'):
    manufacturing_year.append(i.text.split('\n')[0].split('-')[1])
manufacturing_year

[' 2019',
 ' 2014',
 ' 2005',
 ' 2020',
 ' 2020',
 ' 2019',
 ' 2013',
 ' 2017',
 ' 2015',
 ' 2013',
 ' 2015',
 ' 2018',
 ' 2014',
 ' 2019',
 ' 2006',
 ' 2008',
 ' 2021',
 ' 2012',
 ' 2017',
 ' 2015',
 ' 2015',
 ' 2017',
 ' 2018',
 ' 2017',
 ' 2015',
 ' 2022',
 ' 2015',
 ' 2022',
 ' 2018',
 ' 2017',
 ' 2021',
 ' 2018',
 ' 2010',
 ' 2016',
 ' 2021',
 ' 2019',
 ' 2019',
 ' 2015',
 ' 2018',
 ' 2016',
 ' 2011',
 ' 2010',
 ' 2017',
 ' 2010',
 ' 2013',
 ' 2015',
 ' 2013',
 ' 2017',
 ' 2015',
 ' 2022',
 ' 2015',
 ' 2022',
 ' 2018',
 ' 2017',
 ' 2021',
 ' 2018',
 ' 2010',
 ' 2016',
 ' 2021',
 ' 2019',
 ' 2019',
 ' 2015',
 ' 2018',
 ' 2016',
 ' 2011',
 ' 2010',
 ' 2017',
 ' 2010',
 ' 2013',
 ' 2015',
 ' 2013',
 ' 2017',
 ' 2018',
 ' 2016',
 ' 2019',
 ' 2015',
 ' 2018',
 ' 2016',
 ' 2016',
 ' 2019',
 ' 2019',
 ' 2016',
 ' 2018',
 ' 2017',
 ' 2020',
 ' 2018',
 ' 2016',
 ' 2014',
 ' 2018',
 ' 2014',
 ' 2008',
 ' 2014',
 ' 2021',
 ' 2016',
 ' 2006',
 ' 2009',
 ' 2018',
 ' 2016',
 ' 2019',
 ' 2015',


In [103]:
len(manufacturing_year)

1872

In [104]:
#Scraping kilometers driven of each car
driven_kilometers=[]
for i in driver.find_elements(By.XPATH,'//div[@class="prime-features"]'):
    driven_kilometers.append(i.text.split('/')[0])
driven_kilometers

['18,376 kms ',
 '27,000 kms ',
 '86,000 kms ',
 '19,990 kms ',
 '22,000 kms ',
 '60,400 kms ',
 '78,900 kms ',
 '46,000 kms ',
 '80,000 kms ',
 '33,000 kms ',
 '1,25,000 kms ',
 '37,000 kms ',
 '45,000 kms ',
 '42,000 kms ',
 '3,00,000 kms ',
 '1,50,000 kms ',
 '26,000 kms ',
 '1,90,000 kms ',
 '21,753 kms ',
 '29,500 kms ',
 '49,700 kms ',
 '61,500 kms ',
 '21,000 kms ',
 '44,000 kms ',
 '98,000 kms ',
 '12 kms ',
 '66,000 kms ',
 '700 kms ',
 '37,000 kms ',
 '27,000 kms ',
 '9,380 kms ',
 '96,000 kms ',
 '40,000 kms ',
 '1,85,000 kms ',
 '35,000 kms ',
 '65,551 kms ',
 '7,594 kms ',
 '59,466 kms ',
 '20,281 kms ',
 '28,738 kms ',
 '45,000 kms ',
 '68,900 kms ',
 '52,000 kms ',
 '80,000 kms ',
 '1,15,000 kms ',
 '76,000 kms ',
 '83,000 kms ',
 '43,500 kms ',
 '98,000 kms ',
 '12 kms ',
 '66,000 kms ',
 '700 kms ',
 '37,000 kms ',
 '27,000 kms ',
 '9,380 kms ',
 '96,000 kms ',
 '40,000 kms ',
 '1,85,000 kms ',
 '35,000 kms ',
 '65,551 kms ',
 '7,594 kms ',
 '59,466 kms ',
 '20,281 kms

In [105]:
len(driven_kilometers)

1872

In [106]:
#Scraping fuel type of each car
Fuel=[]
for i in driver.find_elements(By.XPATH,'//div[@class="prime-features"]'):
    Fuel.append(i.text.split('/')[1])
Fuel

[' Petrol ',
 ' Diesel ',
 ' Petrol',
 ' Diesel ',
 ' Petrol ',
 ' Diesel ',
 ' Petrol',
 ' Diesel',
 ' Diesel',
 ' Diesel',
 ' Petrol',
 ' Petrol ',
 ' Petrol',
 ' Diesel',
 ' Diesel ',
 ' Diesel',
 ' Diesel ',
 ' Diesel',
 ' Petrol',
 ' Petrol',
 ' Diesel ',
 ' Diesel',
 ' Diesel ',
 ' Petrol ',
 ' Diesel',
 ' Diesel',
 ' Petrol',
 ' Petrol ',
 ' Petrol ',
 ' Petrol',
 ' Petrol ',
 ' Diesel',
 ' Petrol',
 ' Diesel ',
 ' Diesel',
 ' Petrol ',
 ' Diesel ',
 ' Petrol ',
 ' Petrol ',
 ' Petrol ',
 ' Petrol',
 ' Petrol',
 ' Petrol',
 ' Petrol',
 ' Diesel ',
 ' Petrol ',
 ' Diesel ',
 ' Petrol ',
 ' Diesel',
 ' Diesel',
 ' Petrol',
 ' Petrol ',
 ' Petrol ',
 ' Petrol',
 ' Petrol ',
 ' Diesel',
 ' Petrol',
 ' Diesel ',
 ' Diesel',
 ' Petrol ',
 ' Diesel ',
 ' Petrol ',
 ' Petrol ',
 ' Petrol ',
 ' Petrol',
 ' Petrol',
 ' Petrol',
 ' Petrol',
 ' Diesel ',
 ' Petrol ',
 ' Diesel ',
 ' Petrol ',
 ' Diesel ',
 ' Petrol ',
 ' Petrol ',
 ' Petrol ',
 ' Petrol ',
 ' Petrol ',
 ' Petrol ',
 ' Petro

In [107]:
len(Fuel)

1872

In [108]:
#Scraping Price of each car
Price=[]
for i in driver.find_elements(By.XPATH,'//div[@class="price"]'):
    Price.append(i.text)
Price

['₹4,08,099',
 '₹11,50,000',
 '₹95,000',
 '₹11,50,000',
 '₹8,25,000',
 '₹32,60,000',
 '₹4,69,000',
 '₹6,49,999',
 '₹4,44,444',
 '₹8,44,444',
 '₹3,25,000',
 '₹5,25,000',
 '₹2,10,000',
 '₹4,15,000',
 '₹3,60,000',
 '₹2,55,000',
 '₹12,75,000',
 '₹5,80,000',
 '₹2,50,000',
 '₹4,50,000',
 '₹4,25,000',
 '₹4,75,000',
 '₹7,50,000',
 '₹2,75,000',
 '₹5,50,000',
 '₹4,32,000',
 '₹5,75,000',
 '₹5,30,000',
 '₹7,60,000',
 '₹6,00,000',
 '₹7,72,000',
 '₹8,25,000',
 '₹2,00,000',
 '₹5,00,000',
 '₹66,66,666',
 '₹9,44,499',
 '₹16,73,499',
 '₹3,21,699',
 '₹7,94,399',
 '₹3,40,299',
 '₹1,65,000',
 '₹2,80,000',
 '₹6,10,000',
 '₹3,45,000',
 '₹7,21,000',
 '₹5,60,000',
 '₹10,00,000',
 '₹6,40,000',
 '₹5,50,000',
 '₹4,32,000',
 '₹5,75,000',
 '₹5,30,000',
 '₹7,60,000',
 '₹6,00,000',
 '₹7,72,000',
 '₹8,25,000',
 '₹2,00,000',
 '₹5,00,000',
 '₹66,66,666',
 '₹9,44,499',
 '₹16,73,499',
 '₹3,21,699',
 '₹7,94,399',
 '₹3,40,299',
 '₹1,65,000',
 '₹2,80,000',
 '₹6,10,000',
 '₹3,45,000',
 '₹7,21,000',
 '₹5,60,000',
 '₹10,00,000'

In [109]:
len(Price)

1872

In [110]:
#Scraping Locations of each car
Location=[]
for i in driver.find_elements(By.XPATH,'//footer[@class="qc-ads__card--footer"]'):
    Location.append(i.text.split('\n')[1])
Location

['Kochi',
 'Delhi',
 'Mumbai',
 'Chennai',
 'Chennai',
 'Hyderabad',
 'Hyderabad',
 'Delhi',
 'Delhi',
 'Delhi',
 'Kolkata',
 'Kolkata',
 'Kolkata',
 'Delhi',
 'Bangalore',
 'Hyderabad',
 'Vizag',
 'Jaipur',
 'Mumbai',
 'Mumbai',
 'Mumbai',
 'Mumbai',
 'Chennai',
 'Bhilwara',
 'Hyderabad',
 'Bangalore',
 'Kolkata',
 'Chennai',
 'Chennai',
 'Pune',
 'Vadodara',
 'Pune',
 'Kolkata',
 'Tirupur',
 'Delhi',
 'Pune',
 'Chennai',
 'Chennai',
 'Chennai',
 'Chennai',
 'Jaipur',
 'Kolkata',
 'Bangalore',
 'Mumbai',
 'Bangalore',
 'Chennai',
 'Delhi',
 'Chennai',
 'Hyderabad',
 'Bangalore',
 'Kolkata',
 'Chennai',
 'Chennai',
 'Pune',
 'Vadodara',
 'Pune',
 'Kolkata',
 'Tirupur',
 'Delhi',
 'Pune',
 'Chennai',
 'Chennai',
 'Chennai',
 'Chennai',
 'Jaipur',
 'Kolkata',
 'Bangalore',
 'Mumbai',
 'Bangalore',
 'Chennai',
 'Delhi',
 'Chennai',
 'Pune',
 'Pune',
 'Pune',
 'Pune',
 'Pune',
 'Chennai',
 'Pune',
 'Chennai',
 'Pune',
 'Chennai',
 'Pune',
 'Chennai',
 'Pune',
 'Chennai',
 'Chennai',
 'Pune

In [111]:
len(Location)

1872

In [112]:
#scraping the url's of each car as some of the information are hidden inside link
product_urls = []
url=driver.find_elements(By.XPATH,'//a[@class="qc-ads__card"]')   
for i in url:
    product_urls.append(i.get_attribute('href'))                  
product_urls

['https://www.quikr.com/cars/used-other-2019-maruti-suzuki-eeco-18376-kms-driven-in-eloor-kochi/p/362463508',
 'https://www.quikr.com/cars/used-white-2014-audi-a4-2.0-tdi-143bhp-27-000-kms-driven-in-patel-nagar-delhi/p/362466342',
 'https://www.quikr.com/cars/used-black-2005-hyundai-santro-xing-88000-kms-driven-in-malad-west-mumbai/p/362457765',
 'https://www.quikr.com/cars/used-white-2020-mahindra-marazzo-m2-8-str-19990-kms-driven-in-ayanavaram-chennai/p/362462224',
 'https://www.quikr.com/cars/used-blue-2020-hyundai-venue-22-000-kms-driven-in-kilpauk-chennai/p/362187348',
 'https://www.quikr.com/cars/used-2019-skoda-kodiaq-60-400-kms-driven-in-adikmet-hyderabad/p/362461124',
 'https://www.quikr.com/cars/used-white-2013-hyundai-fluidic-verna-1.6-vtvt-sx-78-900-kms-driven-in-adikmet-hyderabad/p/362461094',
 'https://www.quikr.com/cars/used-black-2017-mahindra-xuv-500-w6-46000-kms-driven-in-jangpura-delhi/p/362460902',
 'https://www.quikr.com/cars/used-black-2015-mahindra-tuv300-t6-8000

In [113]:
len(product_urls)

1872

In [132]:
#scraping the data of owners in several parts as the data is large
No_of_Owner1 = []   
for i in product_urls[0:500]:
    driver.get(i)
    time.sleep(1)
    try:
        view=driver.find_element(By.XPATH,'/html/body/div/div[2]/div[1]/div[1]/div[2]/div/div[1]/span')
        view.click()
        ow=driver.find_element(By.XPATH,'//ul[@class="sc-1sbb775-0 lhcfIZ dF fwW jcSB"]')
        No_of_Owner1.append(ow.text.split('\n')[-1])
    except NoSuchElementException:
        No_of_Owner1.append('-')

In [133]:
No_of_Owner2 = []   
for i in product_urls[500:1000]:
    driver.get(i)
    time.sleep(1)
    try:
        view=driver.find_element(By.XPATH,'/html/body/div/div[2]/div[1]/div[1]/div[2]/div/div[1]/span')
        view.click()
        ow=driver.find_element(By.XPATH,'//ul[@class="sc-1sbb775-0 lhcfIZ dF fwW jcSB"]')
        No_of_Owner2.append(ow.text.split('\n')[-1])
    except NoSuchElementException:
        No_of_Owner2.append('-')

In [134]:
No_of_Owner3 = []   
for i in product_urls[1000:1500]:
    driver.get(i)
    time.sleep(1)
    try:
        view=driver.find_element(By.XPATH,'/html/body/div/div[2]/div[1]/div[1]/div[2]/div/div[1]/span')
        view.click()
        ow=driver.find_element(By.XPATH,'//ul[@class="sc-1sbb775-0 lhcfIZ dF fwW jcSB"]')
        No_of_Owner3.append(ow.text.split('\n')[-1])
    except NoSuchElementException:
        No_of_Owner3.append('-')

In [135]:
No_of_Owner4 = []   
for i in product_urls[1500:]:
    driver.get(i)
    time.sleep(1)
    try:
        view=driver.find_element(By.XPATH,'/html/body/div/div[2]/div[1]/div[1]/div[2]/div/div[1]/span')
        view.click()
        ow=driver.find_element(By.XPATH,'//ul[@class="sc-1sbb775-0 lhcfIZ dF fwW jcSB"]')
        No_of_Owner4.append(ow.text.split('\n')[-1])
    except NoSuchElementException:
        No_of_Owner4.append('-')

In [136]:
#final column for total owners
No_of_Owner=No_of_Owner1+No_of_Owner2+No_of_Owner3+No_of_Owner4
No_of_Owner

['One',
 'One',
 '-',
 'One',
 '-',
 '-',
 'Dec 29 2022',
 '-',
 '-',
 '-',
 '-',
 'One',
 '-',
 '-',
 'One',
 '-',
 'One',
 '-',
 '-',
 '-',
 '-',
 '-',
 'One',
 'One',
 '-',
 '-',
 '-',
 'One',
 'One',
 '-',
 'One',
 '-',
 '-',
 'One',
 '-',
 'Two',
 'Two',
 'Two',
 'Two',
 'Two',
 'Manual',
 '-',
 '-',
 'Manual',
 'One',
 'One',
 '-',
 'One',
 '-',
 '-',
 '-',
 'One',
 'One',
 '-',
 'One',
 '-',
 '-',
 'One',
 '-',
 'Two',
 'Two',
 'Two',
 'Two',
 'Two',
 'Manual',
 '-',
 '-',
 'Manual',
 'One',
 'One',
 '-',
 'One',
 'Two',
 'Two',
 'Two',
 'Two',
 'Two',
 'Two',
 'One',
 'Two',
 'Two',
 'Two',
 'Two',
 'Two',
 'Two',
 'Two',
 'Two',
 'Three',
 '-',
 '-',
 'Three',
 '-',
 'One',
 '-',
 '-',
 'Manual',
 'Two',
 'Two',
 'Two',
 'Two',
 'Two',
 'Two',
 'One',
 'Two',
 'Two',
 'Two',
 'Two',
 'Two',
 'Two',
 'Two',
 'Two',
 'Three',
 '-',
 '-',
 'Three',
 '-',
 'One',
 '-',
 '-',
 'Manual',
 'Three',
 'Two',
 'Two',
 'Two',
 'Two',
 'Two',
 'One',
 'Two',
 'Two',
 'Three',
 'Two',
 'On

In [114]:
#same i'm doing for brands too
Brand1 = []   
for i in product_urls[0:500]:
    driver.get(i)
    time.sleep(1)
    try:
        brand = driver.find_element(By.XPATH,'//ul[@class="sc-1sbb775-0 lhcfIZ dF fwW jcSB"]')               
        Brand1.append(brand.text.split('\n')[5])
    except NoSuchElementException:
        Brand1.append('-')

In [116]:
Brand2 = []   
for i in product_urls[500:1000]:
    driver.get(i)
    time.sleep(1)
    try:
        brand = driver.find_element(By.XPATH,'//ul[@class="sc-1sbb775-0 lhcfIZ dF fwW jcSB"]')               
        Brand2.append(brand.text.split('\n')[5])
    except NoSuchElementException:
        Brand2.append('-')

In [117]:
Brand3 = []   
for i in product_urls[1000:1500]:
    driver.get(i)
    time.sleep(1)
    try:
        brand = driver.find_element(By.XPATH,'//ul[@class="sc-1sbb775-0 lhcfIZ dF fwW jcSB"]')               
        Brand3.append(brand.text.split('\n')[5])
    except NoSuchElementException:
        Brand3.append('-')

In [118]:
Brand4 = []   
for i in product_urls[1500:]:
    driver.get(i)
    time.sleep(1)
    try:
        brand = driver.find_element(By.XPATH,'//ul[@class="sc-1sbb775-0 lhcfIZ dF fwW jcSB"]')               
        Brand4.append(brand.text.split('\n')[5])
    except NoSuchElementException:
        Brand4.append('-')

In [119]:
Brand=Brand1+Brand2+Brand3+Brand4
Brand

['Maruti Suzuki',
 'Audi',
 'Hyundai',
 'Mahindra',
 'Hyundai',
 'Skoda',
 'Hyundai',
 'Mahindra',
 'Mahindra',
 'BMW',
 'Maruti Suzuki',
 'Toyota',
 'Maruti Suzuki',
 'Hyundai',
 'Toyota',
 'Maruti Suzuki',
 'Tata',
 'Mitsubishi',
 'Datsun',
 'Maruti Suzuki',
 'Hyundai',
 'Hyundai',
 'Maruti Suzuki',
 'Renault',
 'Nissan',
 'Mahindra',
 'Honda',
 'Maruti Suzuki',
 'Honda',
 'Honda',
 'Maruti Suzuki',
 'Maruti Suzuki',
 'Hyundai',
 'Maruti Suzuki',
 'Land Rover',
 'Hyundai',
 'Tata',
 'Renault',
 'Ford',
 'Renault',
 'Hyundai',
 'Skoda',
 'Maruti Suzuki',
 'Maruti Suzuki',
 'Renault',
 'Ford',
 'Mercedes Benz',
 'Hyundai',
 'Nissan',
 'Mahindra',
 'Honda',
 'Maruti Suzuki',
 'Honda',
 'Honda',
 'Maruti Suzuki',
 'Maruti Suzuki',
 'Hyundai',
 'Maruti Suzuki',
 'Land Rover',
 'Hyundai',
 'Tata',
 'Renault',
 'Ford',
 'Renault',
 'Hyundai',
 'Skoda',
 'Maruti Suzuki',
 'Maruti Suzuki',
 'Renault',
 'Ford',
 'Mercedes Benz',
 'Hyundai',
 'Hyundai',
 'Honda',
 'Honda',
 'Nissan',
 'Honda',


In [120]:
len(Brand)

1872

In [145]:
import pandas as pd
df=pd.DataFrame({'Brand':Brand,'Car Name':name,'Manufacturing Year':manufacturing_year,'Driven KMs':driven_kilometers,'Fuel Type':Fuel,'Car Price':Price,'Location':Location,'Total Owners':No_of_Owner,'URL"s':product_urls})
df

Brand                              Car Name Manufacturing Year  \
0     Maruti Suzuki                   Maruti Suzuki Eeco                2019   
1              Audi               Audi A4 2.0 TDI 143bhp                2014   
2           Hyundai                  Hyundai Santro Xing                2005   
3          Mahindra            Mahindra Marazzo M2 8 STR                2020   
4           Hyundai                        Hyundai Venue                2020   
...             ...                                   ...                ...   
1867        Hyundai    Hyundai i20 Asta 1.4 CRDI 6 Speed                2018   
1868        Hyundai  Hyundai Grand i10 1.2 CRDI Sportz O                2017   
1869           Audi            Audi Q3 2.0 TDI S Edition                2014   
1870           Audi            Audi Q3 2.0 TDI S Edition                2014   
1871        Hyundai            Hyundai Elantra 1.6 SX MT                2013   

       Driven KMs Fuel Type   Car Price    Location Total Owners  \
0     18,376 kms    Petrol    ₹4,08,099       Kochi          One   
1     27,000 kms    Diesel   ₹11,50,000       Delhi          One   
2     86,000 kms     Petrol     ₹95,000      Mumbai            -   
3     19,990 kms    Diesel   ₹11,50,000     Chennai          One   
4     22,000 kms    Petrol    ₹8,25,000     Chennai            -   
...           ...       ...         ...         ...          ...   
1867  69,000 kms     Diesel   ₹7,50,000  Vijayawada            -   
1868  73,600 kms     Petrol   ₹4,20,000    Srinagar            -   
1869  66,027 kms    Diesel   ₹10,89,449     Gurgaon          One   
1870  93,752 kms    Diesel   ₹10,37,699       Noida        Three   
1871  50,000 kms    Diesel    ₹4,25,000     Kolkata        Three   

                                                  URL"s  
0     https://www.quikr.com/cars/used-other-2019-mar...  
1     https://www.quikr.com/cars/used-white-2014-aud...  
2     https://www.quikr.com/cars/used-black-2005-hyu...  
3     https://www.quikr.com/cars/used-white-2020-mah...  
4     https://www.quikr.com/cars/used-blue-2020-hyun...  
...                                                 ...  
1867  https://www.quikr.com/cars/used-grey-2018-hyun...  
1868  https://www.quikr.com/cars/used-white-2017-hyu...  
1869  https://www.quikr.com/cars/used-white-2014-aud...  
1870  https://www.quikr.com/cars/used-white-2014-aud...  
1871  https://www.quikr.com/cars/used-grey-2013-hyun...  

[1872 rows x 9 columns]

In [147]:
df.to_excel('Quikr_used_car.xlsx')

##### I've scraped down 1872 used car informations in this notebook with the help of Quikr website and rest i'll be scraping from other websites in next notebooks.